File: C++(3)/8.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, m;
    cout << "Ingrese número de filas (n): ";
    cin >> n;
    cout << "Ingrese número de columnas (m): ";
    cin >> m;

    vector<vector<int>> matriz(n, vector<int>(m));
    vector<vector<int>> transpuesta(m, vector<int>(n));

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cout << "Elemento [" << i << "][" << j << "]: ";
            cin >> matriz[i][j];
        }
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            transpuesta[j][i] = matriz[i][j];
        }
    }

    cout << "Matriz transpuesta:" << endl;
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            cout << transpuesta[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

File: Py (3)/8.py
def transposicion(matriz):
    n = len(matriz)
    m = len(matriz[0])
    transpuesta = [[0]*n for _ in range(m)]

    for i in range(n):
        for j in range(m):
            transpuesta[j][i] = matriz[i][j]
    return transpuesta

# Programa principal
n = int(input("Ingrese número de filas: "))
m = int(input("Ingrese número de columnas: "))

matriz = []
for i in range(n):
    fila = []
    for j in range(m):
        valor = int(input(f"Elemento [{i}][{j}]: "))
        fila.append(valor)
    matriz.append(fila)

print("Matriz transpuesta:")
for fila in transposicion(matriz):
    print(fila)
